airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -68,11 +68,9 @@ def clear_cache(self):
         """
         remove cache file only once
         """
-        STREAM_CACHE_FILES = globals().setdefault("STREAM_CACHE_FILES", set())
-        if self.cache_filename not in STREAM_CACHE_FILES:
-            with suppress(FileNotFoundError):
-                os.remove(self.cache_filename)
-            STREAM_CACHE_FILES.add(self.cache_filename)
+        with suppress(FileNotFoundError):
+            os.remove(self.cache_filename)
+            print(f"Removed {self.cache_filename}")
```

Comment:
This can be dangerous, for example:

1. We create `Teams` stream, on start it removes cache file and create it - `file-inode-1`
2. We create `TeamMembers(parent=Teams)`, on start it again removes cache file and create it `file-inode-2`
3. `Teams` assume it has access to `file-inode-1` but in realy it's already `file-inode-2`

I definitely remember there were some side-effects if we removed file not once

Reply:
We start to catch sqlite3 runtime exceptions

Reply:
let me cover that runtime side-effects with unit_tests today

Reply:
@sherifnada example of problem from `source-github`:
```
File "/home/user/airbyte/airbyte-integrations/connectors/source-github/.venv/lib/python3.9/site-packages/requests_cache/backends/base.py", line 100, in save_response
    self.responses[cache_key] = cached_response
  File "/home/user/airbyte/airbyte-integrations/connectors/source-github/.venv/lib/python3.9/site-packages/requests_cache/backends/sqlite.py", line 268, in __setitem__
    super().__setitem__(key, serialized_value)
  File "/home/user/airbyte/airbyte-integrations/connectors/source-github/.venv/lib/python3.9/site-packages/requests_cache/backends/sqlite.py", line 220, in __setitem__
    con.execute(
sqlite3.OperationalError: attempt to write a readonly database
```

Reply:
@sherifnada 

I have dived into request_cache internals and it seems it happens something like this under the hood:

```python
#!/home/user/airbyte/airbyte-integrations/connectors/source-github/.venv/bin/python3

import os
import sqlite3

try:
    os.unlink("cache.sqlite")
except FileNotFoundError:
    pass

con = sqlite3.connect("cache.sqlite")
row = con.execute('CREATE TABLE t (col VARCHAR)')
con.commit()

os.unlink("cache.sqlite") # <- ATTENTION HERE

con.execute("INSERT INTO t (col) VALUES ('value')");
```

```python
Traceback (most recent call last):
  File "/home/user/airbyte/airbyte-integrations/connectors/source-github/./t.py", line 28, in <module>
    con.execute("INSERT INTO t (col) VALUES ('value')");
sqlite3.OperationalError: attempt to write a readonly database
```